src/BuildingBlocks/EventBus/EventBus/InMemoryEventBusSubscriptionsManager.cs

```diff
@@ -1,162 +1,155 @@
-﻿using Microsoft.eShopOnContainers.BuildingBlocks.EventBus.Abstractions;
-using Microsoft.eShopOnContainers.BuildingBlocks.EventBus.Events;
-using System;
-using System.Collections.Generic;
-using System.Linq;
+﻿namespace Microsoft.eShopOnContainers.BuildingBlocks.EventBus;
 
-namespace Microsoft.eShopOnContainers.BuildingBlocks.EventBus
+public partial class InMemoryEventBusSubscriptionsManager : IEventBusSubscriptionsManager
 {
-    public partial class InMemoryEventBusSubscriptionsManager : IEventBusSubscriptionsManager
+
+
+    private readonly Dictionary<string, List<SubscriptionInfo>> _handlers;
+    private readonly List<Type> _eventTypes;
+
+    public event EventHandler<string> OnEventRemoved;
+
+    public InMemoryEventBusSubscriptionsManager()
     {
+        _handlers = new Dictionary<string, List<SubscriptionInfo>>();
+        _eventTypes = new List<Type>();
+    }
 
+    public bool IsEmpty => !_handlers.Keys.Any();
```

Comment:
```suggestion
    public bool IsEmpty => _handlers is { Count: 0 };
```

Use pattern matching, and avoid negation logic as it requires a bit more cognitive load to quickly read.

Reply:
Ok, sure